qiskit/providers/ibmq/api_v2/clients/websocket.py

```diff
@@ -203,12 +223,18 @@ def get_job_status(self, job_id, timeout=None):
                         attempt_retry = False  # No point in retrying.
                         message = 'Job id not found'
 
-                    if attempt_retry:
+                    # Attempt to retry.
+                    if current_retry_attempt < retries and attempt_retry:
                         logger.warning('Connection with the websocket closed '
                                        'unexpectedly: %s(status_code=%s). '
                                        'Retrying get_job_status.', message, ex.code)
-                        attempt_retry = False  # Disallow further retries.
+
+                        # Reestablish websocket connection and increment retry counter.
                         websocket = yield from self._connect(url)
+                        current_retry_attempt = current_retry_attempt + 1
+
+                        # Block asyncio loop for a given backoff value.
+                        yield from self._sleep_backoff(backoff_factor, current_retry_attempt)
```

Comment:
Usually you want to sleep first then do the retry (e.g. re-connect). This, for example, gives a restarting server time to finish initialization before reconnect is attempted.